yjit/src/codegen.rs

```diff
@@ -1770,6 +1770,30 @@ fn gen_setlocal_generic(
     // Load environment pointer EP at level
     let ep_opnd = gen_get_ep(asm, level);
 
+    // Fallback because of write barrier
+    if asm.ctx.get_chain_depth() > 0
+    {
+        // Max says: I don't think that this is needed here?
+        // Save the PC and SP because we are allocating
+        //jit_prepare_routine_call(jit, asm);
```

Comment:
```suggestion
        // Save the PC and SP because it runs GC
        jit_prepare_routine_call(jit, asm);
```

`vm_env_write_slowpath` seems to run GC, so I believe it's necessary.